src/main/java/org/opensearch/flowframework/util/ParseUtils.java

```diff
@@ -251,6 +252,31 @@ public static User getUserContext(Client client) {
         return User.parse(userStr);
     }
 
+    /**
+     * Checks whether resource authz framework should be used.
+     * @param resourceType the type to check for resource authz
+     * @return true if resource-authz should be used, false otherwise
+     */
+    public static boolean shouldUseResourceAuthz(String resourceType) {
+        var client = ResourceSharingClientAccessor.getInstance().getResourceSharingClient();
+        return client != null && client.isFeatureEnabledForType(resourceType);
+    }
+
+    /**
+     * Verifies whether the user has permission to access the resource.
+     * @param resourceType the type of resource to be authorized
+     * @param onSuccess consumer function to execute if resource sharing feature is enabled
+     * @param fallbackOn501 consumer function to execute if resource sharing feature is disabled.
+     */
+    public static void verifyResourceAccessAndProcessRequest(String resourceType, Runnable onSuccess, Runnable fallbackOn501) {
```

Comment:
```suggestion
    public static void verifyResourceAccessAndProcessRequest(String resourceType, Runnable onSuccess, Runnable onFailure) {
```

Reply:
it is not on failure. it will only fallback when resource-sharing framework is not used.